[Bug]: make test fails on macOS due to missing gsed (GNU sed) prerequisite

### What happened?

As a new contributor trying to run tests locally,
I want to be able to run `make test` on macOS without failures,
so that I can contribute effectively without debugging environment-specific issues.

Currently, running `make test` on macOS fails with the following error:

```bash
gsed: command not found
make: *** [some-target] Error 127
```

There is no mention in the documentation that `gnu-sed (gsed)` is required, which makes this confusing for new contributors.

### Steps to reproduce

1. Clone the Jaeger repo on macOS (e.g., `git clone https://github.com/jaegertracing/jaeger.git`)
2. Run `make test` without installing GNU `sed`
3. Observe failure with `gsed: command not found`

**NOTE:**  This will only going to fail if `gsed` is not installed on your system. It should be documented in the `CONTRIBUTING.md` file.

### Expected behavior

* Either:

  * `make test` should work out-of-the-box on macOS, or
  * The documentation should clearly state that `gsed` (GNU sed) is required and how to install it:

    ```bash
    brew install gnu-sed
    ```

### Relevant log output

```shell

```

### Screenshot

_No response_

### Additional context

### System config:
**OS:** macOS Sequoia 15.5
**RAM:** 24GB


### Jaeger backend version

_No response_

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

_No response_

### Operating system

_No response_

### Deployment model

_No response_

### Deployment configs

```shell

```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -18,6 +18,13 @@ We gratefully welcome improvements to documentation as well as to code.
 
 This library uses Go modules to manage dependencies.
 
+If you are running `make test` or other Makefile targets on macOS, please ensure that you have GNU `sed` installed.
+
+To install GNU `sed`:
+
+```bash
+brew install gnu-sed
+```
 
 ```
 git clone git@github.com:jaegertracing/jaeger.git jaeger
@@ -49,6 +56,7 @@ make fmt  # commit all changes from auto-format
 make lint
 make test
 ```
+
 
 ### Auto-format
 

